src/m365/entra/commands/app/app-permission-add.ts

```diff
@@ -139,13 +145,50 @@ class EntraAppPermissionAddCommand extends GraphCommand {
     }
   }
 
+  private async getAppObjectByName(appName: string): Promise<Application[]> {
```

Comment:
Hi @SmitaNachan,

In this function you're now executing 2 http requests, while it would be ok to use just one. There's no reason that I see to execute two requests.

In the other function you're also executing a request on the same endpoint. I suggest we move it all in one function, determine the $filter query param based on the available options and then just use one odata.getAllItems call.

Let's consolidate this a bit before we merge it.